[bryancheny] iP

warg-void opened the pull request:
### Crack Chatbot Project

Crack is a simple task management chatbot application that helps users manage their tasks efficiently.

### Features

- _Manage_ tasks with different types: Todo, Deadline, Event
- Save and load tasks from a file
- **User-friendly** interface for interaction
- Command parsing for task operations

### How to Use

1. Initialize the application: Run the main method in Crack.java.
2. Add tasks: Use commands to add Todo, Deadline, or Event tasks.
3. Mark tasks as done: Mark tasks as completed using specific commands.
4. Save tasks: Tasks are automatically saved to a file.
5. Load tasks: Tasks are loaded from the file on startup.

`./gradlew clean build`
`./gradlew run`

### Development Tasks

- [ ] <input disabled="" type="checkbox"> Add more user commands
- [ ] <input disabled="" type="checkbox"> Improve error handling

> "Crack makes task management a breeze!" - Happy User

This is a _fork_ of [this github repository](https://github.com/nus-cs2103-AY2425S1/ip)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

warg-void committed:
Change name and greeting message

warg-void committed:
add bot echoing and exit functionality

warg-void committed:
remove bye message from greeting

warg-void committed:
add ability to store user text

warg-void committed:
add mark as done feature

warg-void committed:
remove class files

warg-void committed:
add todo, events, deadlines

warg-void committed:
correct test script

warg-void committed:
add more error handling cases

warg-void committed:
add delete function and update tests

warg-void committed:
add tasktype enum

warg-void committed:
Implement saving

warg-void committed:
modify gitignore

warg-void committed:
Merge Level 7

warg-void committed:
Add datetimes

warg-void committed:
Merge Level 8

yockcheng commented on src/main/java/Crack.java:
```diff
@@ -0,0 +1,196 @@
+import java.util.Scanner;
+import java.io.File;
```
Should the imports be done in lexicographical order?

yockcheng commented on src/main/java/Crack.java:
```diff
@@ -0,0 +1,196 @@
+                    } else {
+                        System.out.println(divider + " Error: Task number out of range.\n" + divider);
+                    }
+                } catch (Exception e) {
```
Perhaps we can be more specific with the Exception being caught here?

yockcheng commented on src/main/java/Crack.java:
```diff
@@ -0,0 +1,196 @@
+            while (fileScanner.hasNextLine()) {
+                String line = fileScanner.nextLine();
+                String[] parts = line.split(" \\| ");
+                switch (parts[0]) {
```
Should the indentation of the "cases" be in line with "switch"?

yockcheng commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,118 @@
+    public String toString() {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy");
+        return super.toString() + " (from: " + from.format(formatter) + " to: " + to.format(formatter) + ")";
+    }
```
I like the naming of the variables and methods.

yockcheng commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,118 @@
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy");
+        return super.toString() + " (from: " + from.format(formatter) + " to: " + to.format(formatter) + ")";
+    }
+}
```
Perhaps consider adding some javadocs in your code?

yockcheng reviewed (commented):
Overall, the code followed most of the coding conventions. I noted that you decided to write nested classes inside Task currently. Maybe you could consider abstracting them out instead if the classes start to get longer.

chiangqinkang commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,118 @@
+    }
+}
+
+class Event extends Task {
```
I think putting this in another file would make the code neater.

chiangqinkang reviewed (commented):
Apart from some project structure issues, LGTM

warg-void committed:
Refactor with Parser, Storage, TaskList and Ui class

warg-void committed:
Refactor into package

warg-void committed:
Merge branch 'add-gradle-support'

warg-void committed:
Merge gradle branch

warg-void committed:
Add Parser and TaskList tests

warg-void committed:
Add Java docs

warg-void committed:
Tweak slightly to conform to coding standard

warg-void committed:
Add find task functionality

warg-void committed:
Add Checkstyle

warg-void committed:
Add GUI with JavaFX

Previously there was no GUI, only CMD-based user interaction.

GUI is more user friendly and we chose JavaFX as our GUI based on its ease of use.

warg-void committed:
Add assertions to document important assumptions throughout the codebase

Integrated Java's assert statements across multiple classes to enforce important assumptions

Included assertions to verify various task assertions were not null

warg-void committed:
Fix style issues

Before this, there were 2 style errors raised by stylecheck.

Caused gradle build to display fail message

This commit fixes the style issues in Parser.java accordingly.

warg-void committed:
Refactor Crack class to improve code quality and adhere to coding standards

Applied the Single Level of Abstraction Principle (SLAP):

-   Extracted command handling logic from the run() method into dedicated methods such as handleMarkCommand(), handleTodoCommand(), handleDeadlineCommand(), etc.
-   This refactoring reduces the length of the run() method, making it more concise and easier to read.
-   Enhances readability by ensuring the run() method operates at a single level of abstraction.

Updated Method Header Comments to comply with coding standard.

warg-void committed:
Merge pull request #2 from bryancheny/A-Assertions

Merge branch A-Assertions

warg-void committed:
Merge pull request #3 from bryancheny/A-CodeQuality

Merge branch A-CodeQuality

warg-void committed:
Implement B-Snooze feature for task postponement and refactor code

Added Snooze Command (B-Snooze):

    Implemented the ability to snooze or postpone tasks by updating their dates.
    Users can now snooze deadlines or events to postpone them with the snooze command.

Refactored Crack Class:

    Removed the unused run() method since the application is now GUI-based.
    Refactored the getResponse() method to follow the SLAP method.

warg-void committed:
Modify README

README was outdated and so updated it to include how to run and credits.

warg-void committed:
Merge tag 'B-Snooze'

warg-void committed:
Merge branch 'refs/heads/B-Snooze'

warg-void committed:
Update README and add userguide

Previously, there was no user guide and new users could be lost on how Crack works.

- Updated Crack with a new user guide.
- Listed the features of Crack in the user guide.
